src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -95,6 +95,12 @@ internal abstract class TexlFunction : IFunction
         // Return true if the function is asynchronous, false otherwise.
         public virtual bool IsAsync => false;
 
+        // Return true if the function is a delegatable asynchronous, false otherwise.
+        // This means that the function should be a valid node for delegation regardless
+        // of the AllowAsyncDelegation feature, and only means that it'll be a valid node when used
+        // as the left hand side of a dotted name node (i.e. the User function in Power Apps)
+        public virtual bool IsDelegatableAsync => false;
```

Comment:
>IsDelegatableAsync [](http://example.com/codeflow?start=28&length=18)

Why do we need separate flag? Isn't it a combination of IsAsync and IsDelegatable?

Reply:
What do you think a good name would be for an async function that we always allow delegation for?

Reply:
We already have a virtual for it.
IsServerDelegatable

Why not just use that?

Reply:
Because IsServerDelegatable indicates the function itself can be delegated in the query, what this is meant to indicate is that async nodes that are made async specifically because of this function should not break delegation (as we presently do). The idea is that this will be standard behavior at some point but currently we only allow for delegating calls where predicate depends on User() function response. We want to keep this functionality while we test and slowly roll out full async delegation for other functions.

Reply:
So basically not so much that it's delegable or not it's more so should further async nodes no longer be delegable because this is an async function.

Reply:
Decided on overwriting the CallNodeValidationStrategy for the function instead.